java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -4118,13 +4119,43 @@ public Table replaceNulls(ReplacePolicyWithColumn... replacements) {
      * for the memory to be released.
      */
     public ContiguousTable[] contiguousSplitGroups() {
-      return Table.contiguousSplitGroups(
+      ContigSplitGroupByResult ret = Table.contiguousSplitGroups(
           operation.table.nativeHandle,
           operation.indices,
           groupByOptions.getIgnoreNullKeys(),
           groupByOptions.getKeySorted(),
           groupByOptions.getKeysDescending(),
-          groupByOptions.getKeysNullSmallest());
+          groupByOptions.getKeysNullSmallest(),
+          false); // not generate uniq keys table
+      return ret.getGroups();
+    }
+
+    /**
+     * Similar to {@link #contiguousSplitGroups}, return an extra uniq keys table in which
+     * each row is corresponding to a group split.
+     *
+     * Splits the groups in a single table into separate tables according to the grouping keys.
+     * Each split table represents a single group.
+     *
+     * Example, see the example in {@link #contiguousSplitGroups}
+     * The `uniqKeysTable` in ContigSplitGroupByResult is:
+     *    a
+     *    b
+     *  Note: only 2 rows because of only has 2 split groups
+     *
+     * @return The split groups and uniq keys table.
+     */
+    public ContigSplitGroupByResult contiguousSplitGroupsAndGenUniqKeys() {
+      ContigSplitGroupByResult result = Table.contiguousSplitGroups(
+              operation.table.nativeHandle,
+              operation.indices,
+              groupByOptions.getIgnoreNullKeys(),
+              groupByOptions.getKeySorted(),
+              groupByOptions.getKeysDescending(),
+              groupByOptions.getKeysNullSmallest(),
+              true); // generate uniq keys table
+      result.getUniqKeyTable();
```

Comment:
I am wondering if here needs to call `getUniqKeyTable` just for building the table ahead.
The table can be lazily created until it is really required by calling this `getUniqKeyTable`.

Reply:
Done